centoswebpanel.php

```diff
@@ -881,38 +894,90 @@ public function addService(
     public function editService($package, $service, array $vars = null, $parent_package = null, $parent_service = null)
     {
         $row = $this->getModuleRow();
-        $api = $this->getApi($row->meta->host_name, $row->meta->api_key, $row->meta->use_ssl);
+        $api = $this->getApi($row->meta->host_name, $row->meta->port, $row->meta->api_key, $row->meta->use_ssl);
+
+        $params = $this->getFieldsFromInput((array) $vars, $package, true);
+        $service_fields = $this->serviceFieldsToObject($service->fields);
+        $params['server_ips'] = $row->meta->host_name;
+
+        // Default fields using service fields
+        if (!isset($params['domain'])) {
+            $params['domain'] = $service_fields->centoswebpanel_domain;
+        }
+
+        if (!isset($params['pass'])) {
+            $params['pass'] = $service_fields->centoswebpanel_password;
+        }
+
+        if (!isset($params['user'])) {
+            $params['user'] = $service_fields->centoswebpanel_username;
+        }
 
         $this->validateService($package, $vars, true);
 
         if ($this->Input->errors()) {
             return;
         }
 
-        if (isset($vars['centoswebpanel_domain'])) {
+        if (isset($params['centoswebpanel_domain'])) {
```

Comment:
If you're checking `$params`, wouldn't it be `$params['domain']`? Then on line 928 `$params['password']`?

Reply:
Updated